File: catalog/admin/section/pub.php
<?php
defined('AUTH') or die('Restricted access');

include_once $_SERVER['DOCUMENT_ROOT'].'/components/catalog/admin/classes/AdminCatalogSection.php';
$SECTION = new AdminCatalogSection();

$arr['id'] = intval($SITE->d[5]);  // section id
$arr['pub'] = $SITE->d[4] == 'pub' ? 1 : 0;  // pub / unpub

$catalog_id = $SECTION->getSection($arr['id'])['catalog_id'];

$SECTION->pub_unpub($arr);

header('location: /admin/com/catalog/cat/'.$catalog_id);

File: dan/system/catalog/section/pub.py
import sys
sys.path.append('system/catalog/classes')
from system.catalog.classes.Section import Section


def pub(SITE):
    print('FUNCTION -> system-> calalog -> section -> pub/unpub')
    type = SITE.p[2]
    id = SITE.p[3]

    SECTION = Section(SITE)
    cat_id = SECTION.pub(type, id)

    if 'cancel' in SITE.post:
        return {'redirect': '/system/catalog/cat/' + str(cat_id)}
    return {'redirect': '/system/catalog/cat/' + str(cat_id)}